### PR TITLE
Issue #549: Add auto-rename of session title on /auto invocation

### LINKED ISSUE #549: auto: Add auto-rename of session title on /auto invocation
## 背景

`/rename` でセッション名を付けると、`claude --resume` の picker や statusline で「今どの作業をしているか」が一目で分かる。特に `/auto 123` のような autonomous 実行は長時間走るため、複数セッションを並行で持つときに識別性が高い。

しかし現状 `/rename` は手動操作。`/auto N` のように Issue 番号が明示されているケースは自動でリネーム可能なはず。

Claude Code 本体には `UserPromptSubmit` hook が `hookSpecificOutput.sessionTitle` を返すとセッション名を上書きできる機能がある（changelog 既出）。これを利用する。

## 目的

`/auto N` 系のプロンプト送信時、Issue title からセッション名を自動生成し、複数並行セッションの識別性を高める。

## 提案

`UserPromptSubmit` hook を新規追加し、ユーザのプロンプトが `/auto <pattern>` にマッチしたら、Issue title を取得してセッション名を自動設定する。

### 命名規則

セッション名生成ルール：

1. Issue title 先頭の `<component>: ` プレフィックス（`/^[a-zA-Z0-9_/-]+: /`）があれば除去
2. `auto #N: <stripped title>` の形式で連結
3. 結合後の文字列が 50 文字を超える場合、49 文字目までを残して末尾に `…` を付与

### マッチパターンと命名

| プロンプト | セッション名 |
|---|---|
| `/auto 123`（title が `foo: Bar baz` 等の component-prefix 形式） | `auto #123: Bar baz`（prefix 除去） |
| `/auto 123 --patch` 等のフラグ付き | 上と同じ |
| `/auto --resume 123` | `auto #123 (resume): Bar baz` |
| `/auto --batch 5` | `auto batch (5 issues)` |
| `/auto --batch 123 124 125` | `auto batch #123,124,125` |
| 結合後 50 文字超 | 49 文字 + `…` で切り詰め |
| `/auto --help` / 番号なし | リネームしない |

### 実装方針

1. **新規 script**: `scripts/hook-rename-on-auto.sh`
   - stdin の JSON event payload から `prompt` フィールドを抽出
   - `/auto` プレフィックスに対する正規表現マッチで分岐
   - Issue 番号がある場合のみ `gh issue view $N --json title -q .title` で title 取得（失敗時はリネームをスキップしてサイレントに通過）
   - 命名規則（上記）に従って組み立て、50 char truncate を適用
   - `{"hookSpecificOutput": {"sessionTitle": "..."}}` を stdout に出力
   - マッチしなければ空出力（既存挙動を維持）

2. **`.claude/settings.json.template` の `hooks` に `UserPromptSubmit` エントリ追加**（実体 `.claude/settings.json` は `install.sh` が template から生成。git tracked は template のみ）
   - matcher 空、command は `$CLAUDE_PROJECT_DIR/scripts/hook-rename-on-auto.sh`
   - timeout 5000ms（既存 `PermissionRequest` hook と同値）

3. **tests**: bats で hook script の各分岐を検証
   - 番号付き `/auto 123` → 期待される JSON 出力
   - title に `component:` プレフィックスあり/なし → prefix 除去が両ケースで正しく動作
   - 命名結果が 50 char 超 → 49 char + `…` で truncate
   - `/auto --batch 5` / `--batch 123 124 125` / `--resume 123` の各パターン
   - 非マッチプロンプト → 空出力
   - `gh` 失敗時 → 空出力（既存セッション名を保持）

## Non-Goals（本 Issue のスコープ外）

- 他 skill（`/code`, `/spec`, `/review`, `/merge`, `/verify`, `/issue`, `/triage`）への自動リネーム展開。これらは単発・短時間の skill で、リネーム後に別作業に移ったとき残るデメリットの方が大きい。必要なら別 Issue で議論
- skill 完了時のセッション名自動クリア。完了検知（`Stop` hook 等）は複雑かつタイミングが微妙。次回 `/auto M` で自動上書き、それ以外で消したい場合は既存の `/rename` を手動使用で対応

### Pre-merge (auto-verified)

- [x]  `scripts/hook-rename-on-auto.sh` が新規作成されている
- [x]  hook script が `hookSpecificOutput.sessionTitle` を含む JSON を出力する実装になっている
- [x]  `.claude/settings.json.template` に `UserPromptSubmit` hook エントリが追加されている
- [x]  `.claude/settings.json.template` の `UserPromptSubmit` hook が `hook-rename-on-auto.sh` を参照している
- [x]  bats テストファイル `tests/hook-rename-on-auto.bats` が追加されている
- [x]  CI で bats テストが pass している

### Post-merge

- [x] `/auto 123` 実行時、セッション名が `auto #123: <stripped title>` の形式（component prefix 除去 + 50 char truncate 適用）に自動変更される（Claude Code の statusline / `claude --resume` picker で確認） 
- [x] `/auto --batch 5` および `/auto --batch 123 124 125` および `/auto --resume 123` 実行時、それぞれ仕様通りのセッション名になる 
- [x] 通常のプロンプト入力時、hook が動作しても体感できる遅延・余計な出力が無い 
- [x] `gh` 失敗・Issue 不在時に既存セッション名が破壊されない 

## Auto-Resolved Ambiguity Points

- **Hook timeout 値**: 5000ms（`gh issue view` の呼び出しが想定 1-3 秒、余裕を持たせる）。既存 `PermissionRequest` hook と同値で揃える
- **bats テストファイルパス**: `tests/hook-rename-on-auto.bats`。既存 `tests/*.bats` 慣行に従う
- **hook script 配置場所**: `scripts/hook-rename-on-auto.sh`。既存 `scripts/log-permission.sh`（PermissionRequest hook）と同階層
- **truncate 長**: 50 char（statusline / picker での視認性、Issue title 一般長 60-80 char を踏まえた折衷値）
- **対象は `/auto` のみ**: 他 skill 拡張は Non-Goals で明示

## Needs Refinement (defer to /spec)

- **Hook 配置スコープの確定**: project-level `.claude/settings.json.template` のみで完結させるか、wholework plugin 配下に hook を追加して self-hosting 配布構成に乗せるか。後者の場合、`/auto` 以外の skill（`/code`, `/verify` 等）への展開可能性とトレードオフ（ただし Non-Goals で当面除外）。spec で決定
- **install.sh への影響**: template を変更した際、既存ユーザの `.claude/settings.json` を再生成するか、手動マージを促すか。template と既存ファイルの差分検出フローを spec で確定

### PR DESCRIPTION
**Pull request by saito**
## Summary

`UserPromptSubmit` hook (`scripts/hook-rename-on-auto.sh`) を新規追加し、ユーザの入力が `/auto` パターンにマッチした場合に Issue title からセッション名を自動生成する機能を実装。

- `/auto 123` → `auto #123: <stripped title>` (component prefix 除去 + 50 char truncate)
- `/auto --resume 123` → `auto #123 (resume): <stripped title>`
- `/auto --batch 5` → `auto batch (5 issues)`
- `/auto --batch 123 124 125` → `auto batch #123,124,125`
- `gh` 失敗・Issue 不在・`--help`・番号なし → 空出力（既存セッション名を保持）

`.claude/settings.json.template` の `hooks` に `UserPromptSubmit` エントリを追加。`./install.sh` 再実行で既存ユーザに適用される。

bats テスト 13 件追加（全パターンをカバー）。

## Verification (pre-merge)

- `scripts/hook-rename-on-auto.sh` が新規作成されている
- hook script が `hookSpecificOutput.sessionTitle` を含む JSON を出力する実装になっている
- `.claude/settings.json.template` に `UserPromptSubmit` hook エントリが追加されている
- `.claude/settings.json.template` の `UserPromptSubmit` hook が `hook-rename-on-auto.sh` を参照している
- bats テストファイル `tests/hook-rename-on-auto.bats` が追加されている
- CI で bats テストが pass している

## Verification (post-merge)

- `/auto 123` 実行時、セッション名が `auto #123: <stripped title>` に自動変更される
- `/auto --batch 5` / `/auto --batch 123 124 125` / `/auto --resume 123` がそれぞれ仕様通りのセッション名になる
- 通常プロンプト入力時、hook が動作しても遅延・余計な出力がない
- `gh` 失敗・Issue 不在時に既存セッション名が破壊されない

**Note**: 既存ユーザは template 変更後に `./install.sh` を手動再実行する必要があります。

closes #549

### CODE REVIEW

File: scripts/hook-rename-on-auto.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    N=$(echo "$REST" | grep -oE '[0-9]+' | head -1)
+    if [ -z "$N" ]; then
+      exit 0
+    fi
+    RAW=$(gh issue view "$N" --json title -q .title 2>/dev/null) || exit 0
+    [ -z "$RAW" ] && exit 0
+    STRIPPED=$(echo "$RAW" | sed -E 's/^[A-Za-z0-9_/-]+:[[:space:]]*//')
+    TITLE="auto #$N: $STRIPPED"
+    ;;
+esac
```
Comment (saito) [inline scripts/hook-rename-on-auto.sh:62]: **[Edge Case] truncate: `…` は3バイトの UTF-8 文字のため、最終バイト数が51になる可能性があります。**

Severity: CONSIDER（Spec 承認済みの既知妥協点）

`${TITLE:0:49}…` の最終文字列は bash の `${#}` では50文字ですが、実バイト数は52バイト（49 ASCII + 3バイト UTF-8 `…`）になります。Spec Notes で「概ね50文字、稀にバイト境界で切れることがある」と明示されており、現状維持で問題ありません。Claude Code 側のセッションタイトルのバイト制限が判明した場合に再評価してください。

推奨修正: Spec 記載通り現状維持。

**Review by saito — COMMENTED** [review]
## 受け入れ条件の検証結果

| 条件 | 結果 | 備考 |
|------|------|------|
| `scripts/hook-rename-on-auto.sh` が新規作成されている | PASS | — |
| hook script が `hookSpecificOutput.sessionTitle` を含む JSON を出力する実装になっている | PASS | — |
| `.claude/settings.json.template` に `UserPromptSubmit` hook エントリが追加されている | PASS | — |
| `.claude/settings.json.template` の `UserPromptSubmit` hook が `hook-rename-on-auto.sh` を参照している | PASS | — |
| bats テストファイル `tests/hook-rename-on-auto.bats` が追加されている | PASS | — |
| CI で bats テストが pass している | PASS | display name fallback: `Run bats tests`（pass） |

## CI ステータス

| ジョブ名 | ステータス | 備考 |
|---------|-----------|------|
| DCO | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## コードレビュー

### 全般コメント（path/line が特定できない問題）

**[Edge Case] `--batch` フラグ先行ケースのテスト欠落**
- path: null
- line: null

`/auto --batch --foo 5` のようにフラグが `--batch` の直後に来る場合、`NUMS` が空になり `exit 0` する（正しい動作）。ただし、このケースのテストが存在しないため、将来のリファクタリングで挙動が変わっても検知できない。Severity: CONSIDER

推奨修正: bats テストに `--batch` 直後にフラグが来る入力（`/auto --batch --other 5`）が空出力を返すことを確認するケースを追加する（実装変更不要）。

### Issue 数サマリー

- ラインコメントとして投稿: 1件（MUST: 0, SHOULD: 0, CONSIDER: 1）
- 全般コメント（body内）: 1件

---
次: `/merge 550` を実行してください。

**Comment by saito** [comment]
## Review Response Summary


### Claude Review Response（Step 10: 軽量統合レビュー 4観点）

| 観点 | ファイル | 行 | 問題 | 対応 | 対応理由 |
|------|---------|-----|------|------|----------|
| Edge Case | scripts/hook-rename-on-auto.sh | 62 | truncate の `…` は3バイトUTF-8で最終バイト数が52になる | スキップ | Spec Notesで「既知の妥協点」として承認済み。実装変更不要 |
| Edge Case | tests/hook-rename-on-auto.bats | null | `--batch --flag 5` パターンのテストケース欠落 | スキップ | CONSIDER レベルかつ動作仕様は正しい。将来の改善候補 |

### 軽量再チェック結果

- MUST/SHOULD 問題: なし（Step 12 修正作業なし）
- 全受け入れ条件: 6件 PASS
- CI 全チェック: SUCCESS
